int32 robot_id
int32 result